File: Homework/Khang_Nguyen/middle-of-the-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
 var middleNode = function(head) {
    let slow = head
    let fast = head
    while (slow.next !== null) {
        slow = slow.next
        fast = fast.next
        if (fast.next === null) break;
        fast = fast.next
        if (fast.next === null) break;
    }
    return slow
};

File: Homework/Quang_Minh/Lesson1/MiddleOfTheLinkedList.java
public class MiddleOfTheLinkedList {

    public ListNode middleNode(ListNode head) {
        ListNode fastPointer = head;
        ListNode slowPointer = head;
        while(true){
            if(fastPointer == null || fastPointer.next == null) break;
            fastPointer = fastPointer.next.next;
            slowPointer = slowPointer.next;
        }
        return slowPointer;
    }
}
